src/userscores/view/templatesmodel.cpp

```diff
@@ -23,12 +23,11 @@ void TemplatesModel::load()
         }
     }
 
-    m_visibleTemplates = m_allTemplates;
-
     for (const Template& templ: m_allTemplates) {
         m_visibleCategoriesTitles << templ.categoryTitle;
     }
 
+    updateTemplatesByCategory();
     emit categoriesChanged();
     emit templatesChanged();
```

Comment:
templatesChanged and currentTemplateChanged are already called from updateTemplatesByCategory, so you can remove these signals from TemplatesModel::load()